In [30]:
import pandas as pd

**Mean and median income by group of citizenship (population aged 18 and over)**

---
Mean and median income in Euro by citizenship
https://ec.europa.eu/eurostat/databrowser/view/ilc_di15/default/table?lang=en

In [31]:
df = pd.read_csv('../../data/mean_and_median_income_by_citizenship_ilc_di15.csv')

In [32]:
df.head()

DATAFLOW        LAST UPDATE    freq  unit  \
0  ESTAT:ILC_DI15$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual  Euro   
1  ESTAT:ILC_DI15$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual  Euro   
2  ESTAT:ILC_DI15$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual  Euro   
3  ESTAT:ILC_DI15$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual  Euro   
4  ESTAT:ILC_DI15$DEFAULTVIEW(1.0)  21/03/25 23:00:00  Annual  Euro   

                        indic_il  \
0  Median equivalised net income   
1  Median equivalised net income   
2  Median equivalised net income   
3  Median equivalised net income   
4  Median equivalised net income   

                                             citizen    sex               age  \
0  EU27 countries (from 2020) except reporting co...  Total  18 years or over   
1  EU27 countries (from 2020) except reporting co...  Total  18 years or over   
2  EU27 countries (from 2020) except reporting co...  Total  18 years or over   
3  EU27 countries (from 2020) except reporting co...  Total  18 years or over   
4  EU27 countries (from 2020) except reporting co...  Total  18 years or over   

       geo  TIME_PERIOD  OBS_VALUE OBS_FLAG  CONF_STATUS  
0  Albania         2017        NaN        u          NaN  
1  Albania         2018        NaN        u          NaN  
2  Albania         2019        NaN        u          NaN  
3  Albania         2020        NaN        u          NaN  
4  Albania         2021     3454.0        u          NaN

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2151 entries, 0 to 2150
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   DATAFLOW     2151 non-null   object 
 1   LAST UPDATE  2151 non-null   object 
 2   freq         2151 non-null   object 
 3   unit         2151 non-null   object 
 4   indic_il     2151 non-null   object 
 5   citizen      2151 non-null   object 
 6   sex          2151 non-null   object 
 7   age          2151 non-null   object 
 8   geo          2151 non-null   object 
 9   TIME_PERIOD  2151 non-null   int64  
 10  OBS_VALUE    1987 non-null   float64
 11  OBS_FLAG     633 non-null    object 
 12  CONF_STATUS  0 non-null      float64
dtypes: float64(2), int64(1), object(10)
memory usage: 218.6+ KB


OBS_FLAG = 633 meaning 633 out of 2151 values were: estimated or a break in the time series has occured, there was a low reliability or they were provisional 

In [34]:
columns_to_drop = ["OBS_FLAG", "DATAFLOW", "CONF_STATUS", "freq", "incgrp", "unit", "LAST UPDATE"]
df = df.drop(columns=[col for col in columns_to_drop if col in df.columns])

In [35]:
df = df.dropna()

In [36]:
for col in df.columns:
    print(f"Uniques values in {col}: {df[col].unique()}")

Uniques values in indic_il: ['Median equivalised net income']
Uniques values in citizen: ['EU27 countries (from 2020) except reporting country'
 'EU28 countries (2013-2020) except reporting country' 'Foreign country'
 'Reporting country'
 'Non-EU27 countries (from 2020) nor reporting country'
 'Non-EU28 countries (2013-2020) nor reporting country']
Uniques values in sex: ['Total']
Uniques values in age: ['18 years or over']
Uniques values in geo: ['Albania' 'Austria' 'Belgium' 'Bulgaria' 'Switzerland' 'Cyprus' 'Czechia'
 'Germany' 'Denmark'
 'Euro area (EA11-1999, EA12-2001, EA13-2007, EA15-2008, EA16-2009, EA17-2011, EA18-2014, EA19-2015, EA20-2023)'
 'Euro area - 18 countries (2014)' 'Euro area - 19 countries  (2015-2022)'
 'Euro area – 20 countries (from 2023)' 'Estonia' 'Greece' 'Spain'
 'European Union (EU6-1958, EU9-1973, EU10-1981, EU12-1986, EU15-1995, EU25-2004, EU27-2007, EU28-2013, EU27-2020)'
 'European Union - 27 countries (2007-2013)'
 'European Union - 27 countries (from

In [37]:
df = df[~df["geo"].isin(['European Union - 27 countries (from 2020)',
                         'Euro area (EA11-1999, EA12-2001, EA13-2007, EA15-2008, EA16-2009, EA17-2011, EA18-2014, EA19-2015, EA20-2023)',
                         'Euro area - 17 countries (2011-2013)',
                         'European Union (EU6-1958, EU9-1973, EU10-1981, EU12-1986, EU15-1995, EU25-2004, EU27-2007, EU28-2013, EU27-2020)',
                         'European Union - 15 countries (1995-2004)',
                         'European Union - 27 countries (2007-2013)',
                         'European Union - 27 countries (from 2020)',
                         'European Union - 28 countries (2013-2020)'
                         'Euro area - 19 countries  (2015-2022)',
                         'Euro area – 20 countries (from 2023)',
                         'Euro area - 18 countries (2014)',
                         'Euro area - 19 countries  (2015-2022)',
                         'European Union - 28 countries (2013-2020)'
                         ])]

columns_to_drop = ["freq", "indic_de", "statinfo", "age", "unit", 'isced11', "sex", "indic_il"]
df = df.drop(columns=[col for col in columns_to_drop if col in df.columns])

In [38]:
for col in df.columns:
    print(f"Uniques values in {col}: {df[col].unique()}")

Uniques values in citizen: ['EU27 countries (from 2020) except reporting country'
 'EU28 countries (2013-2020) except reporting country' 'Foreign country'
 'Reporting country'
 'Non-EU27 countries (from 2020) nor reporting country'
 'Non-EU28 countries (2013-2020) nor reporting country']
Uniques values in geo: ['Albania' 'Austria' 'Belgium' 'Bulgaria' 'Switzerland' 'Cyprus' 'Czechia'
 'Germany' 'Denmark' 'Estonia' 'Greece' 'Spain' 'Finland' 'France'
 'Croatia' 'Hungary' 'Ireland' 'Iceland' 'Italy' 'Luxembourg' 'Latvia'
 'Montenegro' 'Malta' 'Netherlands' 'Norway' 'Poland' 'Portugal' 'Serbia'
 'Sweden' 'Slovenia' 'Slovakia' 'Türkiye' 'United Kingdom' 'Kosovo*'
 'Lithuania' 'North Macedonia' 'Romania']
Uniques values in TIME_PERIOD: [2021 2015 2016 2017 2018 2019 2020 2022 2023 2024]
Uniques values in OBS_VALUE: [ 3454. 17608. 18004. ... 17904.  8439. 10753.]


In [39]:
df["OBS_VALUE"] = df["OBS_VALUE"].astype(int)

In [40]:
df = df.rename(columns={"OBS_VALUE": "Income", "TIME_PERIOD": "Time period"})

In [41]:
df.to_csv("../../processed_data/mean_and_median_income_by_citizenship_ilc_di15.csv", index=False)